tests/test_forward_ref.py

```diff
@@ -1302,22 +1302,20 @@ def inner():
         func_val(a=1)
 
 
-@pytest.mark.xfail(
-    reason='In `GenerateSchema`, only the current class module is taken into account. '
-    'This is similar to `test_uses_the_correct_globals_to_resolve_model_forward_refs`.'
-)
 def test_uses_the_correct_globals_to_resolve_forward_refs_on_serializers(create_module):
+    # Note: unlike `test_uses_the_correct_globals_to_resolve_model_forward_refs`,
+    # we use the globals of the underlying func to resolve the return type.
     @create_module
     def module_1():
-        from pydantic import BaseModel, field_serializer  # or model_serializer
+        from pydantic import BaseModel, field_serializer  # or model_serializer, computed_field
```

Comment:
```suggestion
        from pydantic import BaseModel, field_serializer  # or model_serializer
```

Reply:
The goal was to mentioned that this also applied to `computed_field`

Reply:
Right, but it's not included in the example?

Reply:
It is commented out, just to mention that the same logic applies to the three decorators. We could add a test for each alternatively.